src/api/entities/SecurityToken/Checkpoints/index.ts

```diff
@@ -37,27 +45,48 @@ export class Checkpoints extends Namespace<SecurityToken> {
   public create: ProcedureMethod<void, Checkpoint>;
 
   /**
-   * Retrieve all Checkpoints created on this Security Token, together with their corresponding creation Date
+   * Retrieve all Checkpoints created on this Security Token, together with their corresponding creation Date and Total Supply
+   *
+   * @note supports pagination
    */
-  public async get(): Promise<CheckpointWithCreationDate[]> {
+  public async get(paginationOpts?: PaginationOptions): Promise<ResultSet<CheckpointWithData>> {
     const {
       parent: { ticker },
       context,
+      context: {
+        polymeshApi: { query },
+      },
     } = this;
 
-    const entries = await context.polymeshApi.query.checkpoint.timestamps.entries(
-      stringToTicker(ticker, context)
-    );
+    const rawTicker = stringToTicker(ticker, context);
+
+    const { entries, lastKey: next } = await requestPaginated(query.checkpoint.totalSupply, {
+      arg: rawTicker,
+      paginationOpts,
+    });
 
-    const now = new Date();
-    return (
-      entries
-        .map(([{ args: [, id] }, timestamp]) => ({
-          checkpoint: new Checkpoint({ id: u64ToBigNumber(id), ticker }, context),
-          createdAt: momentToDate(timestamp),
-        }))
-        // the query also returns the next scheduled checkpoint for every schedule (which haven't been created yet)
-        .filter(({ createdAt }) => createdAt <= now)
+    const checkpoints = entries.map(([{ args: [, id] }, balance]) => ({
+      tuple: tuple(id, rawTicker),
+      id,
+      balance,
+    }));
```

Comment:
You can avoid having to iterate once again over this to create the args for the multi if you use a forEach and two separate arrays

Reply:
Also, there is no reason not to convert id into a Checkpoint and the balance to a BigNumber at this point